conjur/user/user_logic.py

```diff
@@ -35,36 +35,38 @@ def rotate_api_key(self, resource_to_update):
         2. Rotate another users API key
         """
         loaded_credentials = self.extract_credentials_from_credential_store()
-        user_from_credential_store = loaded_credentials['login_id']
-        if resource_to_update == user_from_credential_store:
-            raise InvalidOperationException("To rotate the API key of the currently logged-in user "\
-                                   "use this command without any flags or options")
-        if resource_to_update is not None:
-            user_from_credential_store = resource_to_update
-            logging.debug(f"Rotating API key for '{user_from_credential_store}'")
-
-            new_api_key = self.rotate_other_api_key(user_from_credential_store)
-            logging.debug(f"Successfully rotated API key for '{user_from_credential_store}'")
-        else:
-            # if the user does not provide a user to rotate their API key,
-            # their own API key will be rotated
+        current_logged_in_username = loaded_credentials['login_id']
+
+        # if the user provides their own user id and they are logged in,
+        # then for the sake of a successful outcome, we will ignore the input.
+        # Otherwise, the Conjur server will fail the request
+        if resource_to_update == current_logged_in_username:
+            resource_to_update = None
+        if resource_to_update is None:
             try:
-                logging.debug(f"Rotating API key for '{user_from_credential_store}'")
-                new_api_key = self.rotate_personal_api_key(user_from_credential_store,
+                logging.debug(f"Rotating API key for '{current_logged_in_username}'")
```

Comment:
current and logged in both refer to the same user.
Choose one of them, and the other is redundant.
`current_username` or `logged_in_username`.